packages/virtualized-lists/Lists/VirtualizedList.js

```diff
@@ -712,14 +747,43 @@ class VirtualizedList extends StateSafePureComponent<Props, State> {
       return prevState;
     }
 
+    let maintainVisibleContentPositionAdjustment =
+      prevState.maintainVisibleContentPositionAdjustment;
+    const prevFirstItemKey = prevState.firstItemKey;
+    const newFirstItemKey = VirtualizedList._getItemKey(newProps, 0);
```

Comment:
`minIndexForVisible` is based on ScrollView index of the item, which might be different from the VirtualizedList item index (for example if you have HeaderComponent) so I don't think we should use that value directly. According to the documentation of `minIndexForVisible` it could be used to avoid maintaining scroll position for loading spinners, but I think those will usually be implemented with Header/Footer components of VirtualizedList.

Because of that I think always using the first item is fine in this case.